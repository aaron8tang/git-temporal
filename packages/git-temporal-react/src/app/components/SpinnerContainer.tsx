import React from 'react';
import { style } from 'app/styles';

import { SpinnerImage } from 'app/components/SpinnerImage';

export interface SpinnerContainerProps {
  isSpinning: boolean;
  children: string | JSX.Element | JSX.Element[];
  style?: string | object;
}

const containerStyle = {
  display: 'block',
  height: '100%',
  width: '100%',
  position: 'relative',
};

const backdropStyle = {
  position: 'absolute',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  top: 0,
  left: 0,
  height: '100%',
  width: '100%',
  zIndex: '5',
  backgroundColor: '@colors.background',
  opacity: 0.9,
};

export const SpinnerContainer = (props: SpinnerContainerProps): JSX.Element => {
  return (
    <div style={style(containerStyle, props.style)}>
      {renderSpinner(props.isSpinning)}
      {props.children}
    </div>
  );
};

const renderSpinner = (isSpinning: boolean) => {
  if (!isSpinning) {
    return null;
  }
  return (
    <div style={style(backdropStyle)}>
      <SpinnerImage />
    </div>
  );
};
