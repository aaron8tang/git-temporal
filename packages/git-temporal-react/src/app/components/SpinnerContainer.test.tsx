import React from 'react';
import { shallow } from 'enzyme';

import { SpinnerContainer } from './SpinnerContainer';

describe('components/SpinnerContainer', () => {
  describe('when rendered with isSelected=true', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <SpinnerContainer isSpinning={true}>
          This should be obscured
        </SpinnerContainer>
      );
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered with isSelected=false', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <SpinnerContainer isSpinning={false}>
          This is should not be obscured
        </SpinnerContainer>
      );
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
