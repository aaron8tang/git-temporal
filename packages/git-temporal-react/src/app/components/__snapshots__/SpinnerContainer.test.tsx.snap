// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SpinnerContainer when rendered with isSelected=false it should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "block",
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  This is should not be obscured
</div>
`;

exports[`components/SpinnerContainer when rendered with isSelected=true it should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "block",
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "height": "100%",
        "left": 0,
        "opacity": 0.9,
        "paddingTop": "10%",
        "position": "absolute",
        "textAlign": "center",
        "top": 0,
        "verticalAlign": "middle",
        "width": "100%",
        "zIndex": "5",
      }
    }
  >
    <Component />
  </div>
  This should be obscured
</div>
`;
